plugin/rename.py

```diff
@@ -172,39 +232,95 @@ def _get_relative_path(self, file_path: str) -> str:
-            else:
-                to_render.append('{}:'.format(uri))
+            filename_line = '{}:'.format(self._get_relative_path(file) if scheme == 'file' else uri)
+            to_render.append(filename_line)
+            reference_document.append(filename_line)
             for edit in changes:
-                start = parse_range(edit['range']['start'])
-                if scheme == "file":
-                    line_content = get_line(wm.window, file, start[0])
+                start_row, start_col_utf16 = parse_range(edit['range']['start'])
+                line_content = get_line(wm.window, file, start_row, strip=False) if scheme == 'file' else \
+                    '<no preview available>'
+                start_col = utf16_to_code_points(line_content, start_col_utf16)
+                original_line = ROWCOL_PREFIX.format(start_row + 1, start_col + 1, line_content.strip() + "\n")
+                reference_document.append(original_line)
+                if scheme == "file" and line_content:
+                    end_row, end_col_utf16 = parse_range(edit['range']['end'])
+                    new_text_rows = edit['newText'].split('\n')
+                    new_line_content = line_content[:start_col] + new_text_rows[0]
+                    if start_row == end_row and len(new_text_rows) == 1:
+                        end_col = start_col if end_col_utf16 <= start_col_utf16 else \
+                            utf16_to_code_points(line_content, end_col_utf16)
+                        if end_col < len(line_content):
+                            new_line_content += line_content[end_col:]
+                    to_render.append(
+                        ROWCOL_PREFIX.format(start_row + 1, start_col + 1, new_line_content.strip() + "\n"))
                 else:
-                    line_content = '<no preview available>'
-                to_render.append(" {:>4}:{:<4} {}".format(start[0] + 1, start[1] + 1, line_content))
-            to_render.append("")  # this adds a spacing between filenames
+                    to_render.append(original_line)
         characters = "\n".join(to_render)
         base_dir = wm.get_project_path(self.view.file_name() or "")
         panel.settings().set("result_base_dir", base_dir)
         panel.run_command("lsp_clear_panel")
         wm.window.run_command("show_panel", {"panel": "output.rename"})
-        fmt = "{} changes across {} files.\n\n{}"
         panel.run_command('append', {
-            'characters': fmt.format(total_changes, file_count, characters),
+            'characters': characters,
             'force': True,
             'scroll_to_end': False
         })
+        panel.set_reference_document("\n".join(reference_document))
+        selection = panel.sel()
+        selection.add(sublime.Region(0, panel.size()))
+        panel.run_command('toggle_inline_diff')
+        selection.clear()
+        buttons = pm.rename_panel_buttons
```

Comment:
I would really prefer a dedicated method in PM rather than just accessing its property like that (even if the property doesn't have underscore prefix).